cmd/yggd/registry.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"sync"
+)
+
+var errWorkerRegistered = registryError("cannot add worker; a worker is already registered")
+
+type registryError string
+
+func (e registryError) Error() string { return string(e) }
+
+type registry struct {
+	mu sync.RWMutex
+	mp map[string]*worker
+}
+
+func (r *registry) set(handler string, w *worker) error {
+	if r.mp == nil {
+		r.mp = make(map[string]*worker)
```

Comment:
should be locked here, no? Also the same for lines 37 and 48.

Reply:
Yep. Absolutely. Good catch.